File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
    char[] sArr = s.toCharArray();
    if (sArr.length == 0) {
      return 0;
    }
    int l = 0;
    int r = 1;
    int maxCount = 1;
    Set<Character> seenChar = new HashSet<>();
    seenChar.add(sArr[l]);
    while (r < sArr.length) {
      if (!seenChar.contains(sArr[r])) {
        seenChar.add(sArr[r]);
        maxCount = Math.max(maxCount, seenChar.size());
        r++;
      } else {
        while (seenChar.contains(sArr[r])) {
          seenChar.remove(sArr[l]);
          l++;
        }
        seenChar.add(sArr[r]);
        r++;
      }
    }
    return maxCount;
    }
}

File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.js
/**
 * @param {string} s
 * @return {number}
 */
var lengthOfLongestSubstring = function(s) {
  if (s.length == 0) return 0;
  let l = 0;
  let r = 1;
  let maxCounter = 1;
  const seenChar = new Set();
  seenChar.add(s[l]);
  while (r < s.length) {
    if (!seenChar.has(s[r])) {
      seenChar.add(s[r]);
      r++;
      maxCounter = Math.max(maxCounter, seenChar.size);
    } else {
      while (seenChar.has(s[r])) {
        seenChar.delete(seenChar.delete(s[l]));
        l++;
      }
      seenChar.add(s[r]);
      r++;
    }
  }
  return maxCounter;
};